.pre-commit-config.yaml

```diff
@@ -36,11 +36,12 @@ repos:
         language: system
         require_serial: true
         args: ["--filter-files"]
-      - id: generate-graphql-ts-types
-        name: Generate GraphQL Typescript definitions
-        entry: docker-compose
-        types: [graphql]
-        language: system
-        require_serial: true
-        pass_filenames: false
-        args: ["run", "--rm", "front", "pnpm", "generate-graphql-ts-types"]
+      # TODO how do I change this to run at the project root folder?
+      # - id: generate-graphql-ts-types
+      #   name: Generate GraphQL Typescript definitions
+      #   entry: docker-compose
+      #   types: [graphql]
+      #   language: system
+      #   require_serial: true
+      #   pass_filenames: false
+      #   args: ["run", "--rm", "front", "pnpm", "graphql-gen"]
```

Comment:
@fhenrich33 is it even necessary with tada to generate types? Which command would you run with pure pnpm without any docker?

Reply:
My intention is that whenever the schema on the BE side changes, I want to run `pnpm graphql-gen` to generate both the unified schema and the introspected definition types.

Reply:
We want these to be in sync with the BE changes so the inferred types are aligned with the updated schema:

![image](https://github.com/user-attachments/assets/d08104d3-9cf8-41c0-ba56-ab2460307ec0)

Reply:
If there's a way to have a unified schema.graphql on the BE side, even better!

Reply:
So we want to have a unified schema (that we can get through `cat`), and then run the `pnpm gql-tada ...` command that requires the authz token?

We can't do a single schema on the BE because we use it for two different services (one without mutations), so it's convenient to have modular files. (Maybe it changes in the future)

Reply:
I see, thanks for the explanation!

We can't really use `cat` by itself as of now since there are two `type Query` instances on the concatenated schema, of which only the last will be picked up by gql.tada. 

Hence why I do the commands described in `pnpm graphql-gen` in the root folder:

```sh
pnpm gql-tada generate schema 'http://localhost:5005/graphql' --header "Authorization: bearer `docker compose exec webapp ./back/fetch_token --test`" --output ./graphql/generated/schema.graphql && pnpm gql-tada generate output
```

To generate a unified schema from the introspection of the http://localhost:5005/graphql local server URL (the one with mutations).

So with all that said, I'd like a way to run `pnpm graphql-gen` from `.pre-commit-config.yaml` whenever the schema changes on the backend.

Reply:
Hmmm, some types get skewed doing that... But I'm okay with the first bit of getting the unified schema as portraited in the package.json, the missing piece is to update that and the `graphql/generated/graphql-env.d.ts` output file when there's a *.graphql schema change from a Backend commit or PR. Would that be possible to do this on CI as you mentioned?

Reply:
> when there's a *.graphql schema change from a Backend commit or PR. Would that be possible to do this on CI as you mentioned?

I think that's feasable with the pre-commit tool in the BE, and an extra script. Should I set something up?

Reply:
okay if I push it to this branch?

Reply:
Yes, please!

Reply:
done :heavy_check_mark: